extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -122,82 +125,84 @@ public function signupForm() {
-			$this->errParam = $response->getVal( 'errParam', '' );
-
-			if ( $this->result == 'ok' ) {
-
-				/*
-				 * Remove when SOC-217 ABTest is finished
-				 */
-				$signupForm = new UserLoginForm( $this->wg->request );
-
-				if ( $signupForm->isAllowedRegisterUnconfirmed() ) {
-					$user = User::newFromName( $this->username );
-					// Get and clear redirect page
-					$userSignupRedirect = $user->getOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME );
-					$user->setOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME, null );
-
-					$user->saveSettings();
-
-					// redirect user
-					if ( !empty( $userSignupRedirect ) ) {
-						// Redirect user to the point where he finished (when signup on create wiki)
-						$title = SpecialPage::getTitleFor( 'CreateNewWiki' );
-						$query = $userSignupRedirect;
-					} else {
-						$title = $user->getUserPage();
-						$query = '';
-					}
-
-					$redirectUrl = $title->getFullURL( $query );
+	public function handleSignupFormSubmit() {
+		$response = $this->app->sendRequest( 'UserSignupSpecial', 'signup' );
+		$result = $response->getVal( 'result', '' );
+
+		if ( $result == 'ok' ) {
+			/*
+			 * Remove when SOC-217 ABTest is finished
+			 */
+			$signupForm = new UserLoginForm( $this->wg->request );
+
+			if ( $signupForm->isAllowedRegisterUnconfirmed() ) {
+				$user = User::newFromName( $this->username );
+				// Get and clear redirect page
+				$userSignupRedirect = $user->getOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME );
+				$user->setOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME, null );
+
+				$user->saveSettings();
+
+				// redirect user
+				if ( !empty( $userSignupRedirect ) ) {
+					// Redirect user to the point where he finished (when signup on create wiki)
+					$title = SpecialPage::getTitleFor( 'CreateNewWiki' );
+					$query = $userSignupRedirect;
 				} else {
+					$title = $user->getUserPage();
+					$query = '';
+				}
+
+				$redirectUrl = $title->getFullURL( $query );
+			} else {
 				/*
 				 * end remove
 				 */
-					$params = [
-						'method' => 'sendConfirmationEmail',
-						'username' => $this->username,
-						'byemail' => intval( $this->byemail ),
-					];
-					$redirectUrl = $this->wg->title->getFullUrl( $params );
-				}
-
-				$this->wg->out->redirect( $redirectUrl );
+				$params = [
+					'method' => 'sendConfirmationEmail',
+					'username' => $this->username,
+					'byemail' => intval( $this->byemail ),
+				];
+				$redirectUrl = $this->wg->title->getFullUrl( $params );
 			}
 
+			$this->track( 'signup-successful' );
+			$this->wg->out->redirect( $redirectUrl );
+		} else {
+			$this->response->setData( [
+				'result' => $result,
+				'msg' => $response->getVal( 'msg', '' ),
+				'errParam' => $response->getVal( 'errParam', '' ),
+			] );
 		}
 	}
 
+	/**
+	 * Returns captcha HTML
+	 */
 	public function captcha() {
-		$this->rawHtml = '';
-		$this->isFancyCaptcha = false;
```

Comment:
Correct, that that flag had been used in the UserSignupSpecial_captcha.php template. It was used to determine if we wrapped that captcha in a <div> with a class of "fancy-captcha". I removed it from the template [here](https://github.com/Wikia/app/pull/6466/files#diff-45e7b8c6bc3ee4ab9fe072f18a7356e0L3), and updated getForm on FancyCaptcha to take care of that div, but it looks like I forgot to remove it from here.